test/src/test/java/org/corfudb/infrastructure/logreplication/LogReplicationFSMTest.java

```diff
@@ -683,51 +677,39 @@ public void cancelSnapshotSyncInProgressAndRetry() throws Exception {
         // Write to Stream will write to some addresses.  SnapshotReader should only read from those addresses
         ((TestSnapshotReader) snapshotReader).setSeqNumsToRead(seqNums);
 
-        // Initial acquire of semaphore, the transition method will block until a transition occurs
-        transitionAvailable.acquire();
+        log.info("**** Transition to Snapshot Sync");
 
-        log.debug("**** Transition to Snapshot Sync");
         // Transition #1: Snapshot Sync Request
-        transition(LogReplicationEventType.SNAPSHOT_SYNC_REQUEST, LogReplicationStateType.IN_SNAPSHOT_SYNC);
+        fsm.input(new LogReplicationEvent(LogReplicationEventType.SNAPSHOT_SYNC_REQUEST));
 
-        // We observe the number of transmitted messages and force a REPLICATION_STOP, when 2 messages have been sent
-        // so we verify the state moves to INITIALIZED again.
-        transitionAvailable.acquire();
-        log.debug("**** Stop Replication");
-        fsm.input(new LogReplicationEvent(LogReplicationEventType.REPLICATION_STOP));
+        // Wait till the fsm reaches IN_SNAPSHOT_SYNC state
+        while (fsm.getState().getType() != LogReplicationStateType.IN_SNAPSHOT_SYNC) {
+            log.debug("Wait for the FSM to reach IN_SNAPSHOT_SYNC state");
```

Comment:
Even though this is a test file, perhaps a small sleep to avoid log flooding in case we are debugging?